.github/workflows/ci.yml

```diff
@@ -177,30 +162,45 @@ jobs:
 
       - uses: taiki-e/install-action@cargo-hack
 
-      - name: Build
+      - name: Build & test for all BPF architectures
         env:
-          CARGO_CFG_BPF_TARGET_ARCH: ${{ matrix.bpf_target_arch }}
-        run: |
-          set -euxo pipefail
-          cargo +nightly hack build \
-            --target ${{ matrix.target }} \
-            -Z build-std=core \
-            --package aya-ebpf \
-            --package aya-log-ebpf \
-            --feature-powerset
-
-      - name: Test
-        env:
-          CARGO_CFG_BPF_TARGET_ARCH: ${{ matrix.bpf_target_arch }}
           RUST_BACKTRACE: full
         run: |
-          set -euxo pipefail
-          cargo hack test \
-            --doc \
-            --package aya-ebpf \
-            --package aya-log-ebpf \
-            --feature-powerset
-
+          set -euo pipefail
+
+          failures=()
+
+          for arch in aarch64 arm loongarch64 mips powerpc64 riscv64 s390x x86_64; do
```

Comment:
nice, add a comment that explains that this vs matrix saves time?

Reply:
Done.